extensions/Worker.Extensions.SignalRService/src/Models/SignalRInvocationContext.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// The context for a SignalR invocation.
+    /// </summary>
+    public sealed class SignalRInvocationContext
+    {
+        /// <summary>
+        /// The arguments of the invocation.
+        /// </summary>
+        public object[]? Arguments { get; set; }
+
+        /// <summary>
+        /// The error message of connections disconnected event.
+        /// Only connections disconnected event can have this property, and it can be empty if the connection is closed with no error.
+        /// </summary>
+        public string Error { get; set; }
+
+        /// <summary>
+        /// The category of the invocation.
+        /// </summary>
+        public SignalRInvocationCategory Category { get; set; }
+
+        /// <summary>
+        ///  For <see cref="SignalRInvocationCategory.messages"/> category, the event is the target in invocation message sent from clients. 
+        ///  For <see cref="SignalRInvocationCategory.connections"/> category, only "connected" and "disconnected" are used.
+        /// </summary>
+        public string Event { get; set; }
+
+        /// <summary>
+        /// The hub the invocation belongs to.
+        /// </summary>
+        public string Hub { get; set; }
+
+        /// <summary>
+        /// The connection ID of the client which triggers the invocation.
+        /// </summary>
+        public string ConnectionId { get; set; }
+
+        /// <summary>
+        /// The user ID of the client which triggers the invocation.
+        /// </summary>
+        public string UserId { get; set; }
+
+        /// <summary>
+        /// The headers of request.
+        /// Headers with duplicated key will be joined by comma.
```

Comment:
>will be joined by comma.

Or should we consider using `<string,stringvalues>`  as the type?

Reply:
I think it is a good idea, but deserialization is still a problem. If we customize the deserialization, how to avoid the customization changed by customers.

Reply:
Would be nice (not critical) to match how `HttpRequestData` exposes this information for consistency, but that uses `HttpHeadersCollection`

Reply:
`HttpHeadersCollection` is in a .NET 5 project, and cannot be referenced by .NET Standard 2.0 project.